streamer/src/nonblocking/quic.rs

```diff
@@ -566,9 +566,12 @@ async fn handle_connection(
                         let last_update = last_update.clone();
                         tokio::spawn(async move {
                             let mut maybe_batch = None;
+                            let exit_check_interval =
+                                (wait_for_chunk_timeout_ms / 10).clamp(10, 1000);
+                            let mut start = Instant::now();
                             while !stream_exit.load(Ordering::Relaxed) {
                                 if let Ok(chunk) = tokio::time::timeout(
-                                    Duration::from_millis(wait_for_chunk_timeout_ms),
+                                    Duration::from_millis(exit_check_interval),
```

Comment:
Have we verified that if this times out, when we call stream.read_chunk again, it will try reading the same chunk? Have we tested this doesn't result in regressions in the case where we actually hit this timeout?

Reply:
This loop and on timeout trying to receive the chunk again is not new. The stream being splitter into multiple chunks and reassembled are old logic and it is already happening in production. See the code before https://github.com/solana-labs/solana/pull/28513. If times out reading a chunk, it will try to read a chunk again. There is not the concept of the "same" chunk though -- we did not receive a chunk and hence the timeout.